### PR TITLE
Generalize PodStore

### PR DESCRIPTION
**Pull request by wojtek-t**
@hantaowang - FYI

### CODE REVIEW

File: clusterloader2/pkg/measurement/util/object_store.go [hunk 1]
```diff
@@ -34,56 +35,70 @@ import (
...
+}
+
+// Stop stops ObjectStore watch.
+func (s *ObjectStore) Stop() {
+	close(s.stopCh)
+}
+
+// PodStore is a convenient wrapper around cache.Store.
+type PodStore struct {
+	objectStore
```
Comment (krzysied) [inline clusterloader2/pkg/measurement/util/object_store.go]: s/objectStore/ObjectStore/

File: clusterloader2/pkg/measurement/util/object_store.go [hunk 2]
```diff
@@ -34,56 +35,70 @@ import (
...
-		return nil, err
+	return &PodStore{
+		ObjectStore: newObjectStore(&v1.Pod{}, lw, selector),
 	}
-	return &PodStore{Store: store, stopCh: stopCh, Reflector: reflector}, nil
 }
 
 // List returns to list of pods (that satisfy conditions provided to NewPodStore).
 func (s *PodStore) List() []*v1.Pod {
 	objects := s.Store.List()
```
Comment (krzysied) [inline clusterloader2/pkg/measurement/util/object_store.go]: s/Store/ObjectStore.Store/

File: clusterloader2/pkg/measurement/util/object_store.go [hunk 3]
```diff
@@ -34,56 +35,70 @@ import (
...
-	pods := make([]*v1.Pod, 0)
+	pods := make([]*v1.Pod, 0, len(objects))
 	for _, o := range objects {
 		pods = append(pods, o.(*v1.Pod))
 	}
 	return pods
 }
-
-// Stop stops podstore watch.
-func (s *PodStore) Stop() {
```
Comment (krzysied) [inline clusterloader2/pkg/measurement/util/object_store.go]: Do we need Stop() or one from ObjectStore will be used?

File: clusterloader2/pkg/measurement/util/object_store.go (hunk 1 above)
Comment (wojtek-t) [reply clusterloader2/pkg/measurement/util/object_store.go]: done

File: clusterloader2/pkg/measurement/util/object_store.go (hunk 2 above)
Comment (wojtek-t) [reply clusterloader2/pkg/measurement/util/object_store.go]: I don't think that's needed.

**Comment by hantaowang** [comment]
> @hantaowang - FYI

I can open another PR afterwards to change PVCStore to ObjectStore.

File: clusterloader2/pkg/measurement/util/object_store.go [hunk 4]
```diff
@@ -34,56 +35,73 @@ import (
...
-type PodStore struct {
+// ObjectStore is a convenient wrapper around cache.Store.
+type ObjectStore struct {
 	cache.Store
 	stopCh    chan struct{}
 	Reflector *cache.Reflector
 }
 
-// NewPodStore creates PodStore based on given namespace, label selector and field selector.
+// newObjectStore creates ObjectStore based on given object  selector.
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/util/object_store.go]: NIT: double space between object and selector

Comment (wojtek-t) [reply clusterloader2/pkg/measurement/util/object_store.go]: Will fix in subsequent PR as this apparently already merged :)

**Pull request merged**